File: services/frontend/src/types/Employer.ts
/**
 * Employer portal domain types — departments, employees, contributions, retirements.
 * Consumed by: employer-demo-data.ts, EmployerDashboard, EmployeeRoster, ContributionReporting, RetirementCoordination
 * Depends on: Nothing (self-contained type definitions)
 */

export interface Department {
  dept_id: string
  name: string
  code: string
  employee_count: number
  monthly_payroll: number
  contact_name: string
  contact_email: string
}

export interface EmployerEmployee {
  member_id: string
  first_name: string
  last_name: string
  department: string
  tier: number
  hire_date: string
  status: 'active' | 'terminated' | 'retired' | 'deceased'
  monthly_salary: number
  retirement_date?: string
  retirement_status?: 'none' | 'pending' | 'approved'
  years_of_service: number
}

export interface ContributionReport {
  report_id: string
  period: string
  department: string
  employee_count: number
  total_employee_contributions: number
  total_employer_contributions: number
  total_gross_payroll: number
  status: 'draft' | 'submitted' | 'verified' | 'discrepancy'
  discrepancies?: ContributionDiscrepancy[]
  submitted_at?: string
}

export interface ContributionDiscrepancy {
  member_id: string
  member_name: string
  field: string
  expected: number
  actual: number
  severity: 'warning' | 'error'
  message: string
}

export interface PendingRetirement {
  member_id: string
  member_name: string
  tier: number
  department: string
  retirement_date: string
  application_status: string
  documents_complete: boolean
  last_day_worked?: string
  estimated_benefit?: number
  application_submitted_at?: string  // ISO timestamp when application was submitted
}

export interface EmployerDashboardStats {
  active_employees: number
  pending_retirements: number
  monthly_payroll: number
  avg_service_years: number
  contribution_rate_employee: number
  contribution_rate_employer: number
}

// ─── Contribution Upload Types ──────────────────────────────────────────────

/** Single row from an employer contribution CSV file */
export interface ContributionFileRow {
  member_id: string
  ssn_last4: string
  name: string
  pay_period_begin: string
  pay_period_end: string
  department: string
  job_classification: string
  gross_earnings: number
  pensionable_earnings: number
  employee_contribution: number
  employer_contribution: number
  employment_status: 'active' | 'terminated' | 'leave'
  transaction_type: 'regular' | 'adjustment' | 'retroactive'
  tier: number
}

/** Single validation issue on a row */
export interface ValidationIssue {
  issue_id: string
  field: string
  severity: 'warning' | 'error'
  message: string
  expected?: string
  actual?: string
  resolved: boolean
}

/** Per-row validation result */
export interface RowValidationResult {
  row_index: number
  status: 'clean' | 'warning' | 'error'
  issues: ValidationIssue[]
  corrections: Record<string, string | number>
  acknowledged_warnings: string[]
}

/** File metadata extracted at upload time */
export interface ContributionFileMetadata {
  file_name: string
  file_size_bytes: number
  row_count: number
  period: string
  department: string
  uploaded_at: string
}

/** Aggregate validation stats */
export interface ValidationSummary {
  total_rows: number
  clean_rows: number
  warning_rows: number
  error_rows: number
  total_issues: number
  total_payroll: number
  total_ee_contributions: number
  total_er_contributions: number
}

File: services/connector/internal/models/employer.go
// Package models — employer domain types for the DERP connector service.
// Maps legacy database tables (DEPARTMENT_REF, MEMBER_MASTER, CONTRIBUTION_HIST)
// to clean employer-facing domain objects.
// Consumed by: employer_handlers.go (connector API), intelligence connector client
// Depends on: legacy DB schema (DEPARTMENT_REF, MEMBER_MASTER, CONTRIBUTION_HIST, CASE_HIST)
package models

// Department represents a city department with aggregate employee/payroll data.
// Mapped from DEPARTMENT_REF joined with MEMBER_MASTER aggregates.
type Department struct {
	DeptID        string  `json:"dept_id"`
	Name          string  `json:"name"`
	Code          string  `json:"code"`
	EmployeeCount int     `json:"employee_count"`
	MonthlyPayroll float64 `json:"monthly_payroll"`
	ContactName   string  `json:"contact_name"`
	ContactEmail  string  `json:"contact_email"`
}

// EmployerEmployee represents a member as seen from the employer perspective.
// Includes computed fields (tier, years of service) derived from MEMBER_MASTER columns.
type EmployerEmployee struct {
	MemberID         string  `json:"member_id"`
	FirstName        string  `json:"first_name"`
	LastName         string  `json:"last_name"`
	Department       string  `json:"department"`
	Tier             int     `json:"tier"`
	HireDate         string  `json:"hire_date"`
	Status           string  `json:"status"`
	MonthlySalary    float64 `json:"monthly_salary"`
	RetirementDate   *string `json:"retirement_date,omitempty"`
	RetirementStatus string  `json:"retirement_status"`
	YearsOfService   float64 `json:"years_of_service"`
}

// ContributionReport represents an aggregate contribution report for a department/period.
// Aggregated from CONTRIBUTION_HIST grouped by period and department.
type ContributionReport struct {
	ReportID                   string  `json:"report_id"`
	Period                     string  `json:"period"`
	Department                 string  `json:"department"`
	EmployeeCount              int     `json:"employee_count"`
	TotalEmployeeContributions float64 `json:"total_employee_contributions"`
	TotalEmployerContributions float64 `json:"total_employer_contributions"`
	TotalGrossPayroll          float64 `json:"total_gross_payroll"`
	Status                     string  `json:"status"`
	SubmittedAt                *string `json:"submitted_at,omitempty"`
}

// EmployerDashboardStats provides aggregate statistics for the employer dashboard.
// Computed from MEMBER_MASTER and CASE_HIST queries.
type EmployerDashboardStats struct {
	ActiveEmployees          int     `json:"active_employees"`
	PendingRetirements       int     `json:"pending_retirements"`
	MonthlyPayroll           float64 `json:"monthly_payroll"`
	AvgServiceYears          float64 `json:"avg_service_years"`
	ContributionRateEmployee float64 `json:"contribution_rate_employee"`
	ContributionRateEmployer float64 `json:"contribution_rate_employer"`
}
